src/components/gd/Button/ActionOrSwitchButton.tsx

```diff
@@ -30,11 +31,13 @@ export const ActionOrSwitchButton = ({
 }) => {
     const toggleNetworkModal = useNetworkModalToggle()
     const { i18n } = useLingui()
-    const { chainId, account } = useActiveWeb3React()
+    const { chainId } = useActiveWeb3React()
     // useEffect(() => { }, [chainId, account])
     if (
-        (chainId === (SupportedChainId.FUSE as number) && requireNetwork === DAO_NETWORK.FUSE) ||
-        (chainId !== (SupportedChainId.FUSE as number) && requireNetwork === DAO_NETWORK.MAINNET)
+        (chainId === (SupportedChainId.FUSE as number) && requireNetwork === DAO_NETWORK.FUSE ) ||
+        (chainId !== (SupportedChainId.FUSE as number) && requireNetwork === DAO_NETWORK.MAINNET) ||
```

Comment:
this line will be true for CELO.
this logic can probably be simplified, by converting requireNetwork to the relevant chainIds
DAO_NETWORK.FUSE = 122
DAO_NETWORK.CELO = 42220
DAO_NETWORK.MAINNET = 1,3,42
then check `ids.includes(chainId)`

Reply:
As explained in slack, this button is for a action on a specific network/chain.
so I believe your proposed solution wouldn't work.
But I do think I have now committed a simplified version